File: dir_to_dict.py
import os
import json

def dir_to_dict(path):
    tree = {}
    def helper(current_path, current_dict):
        try:
            for entry in os.listdir(current_path):
                full_path = os.path.join(current_path, entry)
                if os.path.isdir(full_path) and entry.isupper():
                    current_dict[entry] = {}
                    helper(full_path, current_dict[entry])
        except PermissionError:
            print(f"Permission denied: {current_path}")
            current_dict["error"] = "Permission denied"
        except Exception as e:
            print(f"An error occurred: {e}")
            current_dict["error"] = str(e)
    helper(path, tree)
    return tree

def main():
    path = "D:/"  # Remplacer par le chemin du répertoire que vous souhaitez parcourir
    tree = dir_to_dict(path)
    with open("filesystem.json", "w", encoding='utf-8') as f:
        json.dump(tree, f, indent=4, ensure_ascii=False)

if __name__ == "__main__":
    main()

File: dirToDict.js
const fs = require('fs');
const path = require('path');

function dirToDict(currentPath) {
    let tree = {};

    function helper(currentPath, currentDict) {
        try {
            const entries = fs.readdirSync(currentPath, { withFileTypes: true });
            for (const entry of entries) {
                if (entry.isDirectory() && entry.name === entry.name.toUpperCase()) {
                    currentDict[entry.name] = {};
                    helper(path.join(currentPath, entry.name), currentDict[entry.name]);
                }
            }
        } catch (err) {
            if (err.code === 'EACCES') {
                console.error(`Permission denied: ${currentPath}`);
                currentDict["error"] = "Permission denied";
            } else {
                console.error(`An error occurred: ${err.message}`);
                currentDict["error"] = err.message;
            }
        }
    }

    helper(currentPath, tree);
    return tree;
}

function main() {
    const dirPath = "E:/";  // Remplacer par le chemin du répertoire que vous souhaitez parcourir
    const tree = dirToDict(dirPath);
    fs.writeFileSync("filesystem.json", JSON.stringify(tree, null, 4), 'utf8');
}

main();
